polyfactory/factories/pydantic_factory.py

```diff
@@ -10,24 +10,35 @@
     Mapping,
     TypeVar,
     cast,
+    Literal,
 )
 
 from polyfactory.exceptions import MissingDependencyException
 from polyfactory.factories.base import BaseFactory
-from polyfactory.field_meta import Constraints, FieldMeta, Null
+from polyfactory.field_meta import FieldMeta, Null, Constraints
 from polyfactory.utils.helpers import unwrap_new_type
 
 try:
-    from pydantic import (
-        BaseModel,
-    )
-    from pydantic.fields import DeferredType, ModelField, Undefined
+    from pydantic import BaseModel
+
 except ImportError as e:
     raise MissingDependencyException("pydantic is not installed") from e
 
+try:
+    from pydantic.fields import DeferredType, ModelField, Undefined  # type: ignore[attr-defined]
+
+    pydantic_version: Literal[1, 2] = 1
```

Comment:
```suggestion
    from pydantic.version import VERSION as pydantic_version
```

?

Reply:
I rather keep it simple. they store the version as a string and i will have to parse it.